Proper configuration system (#9)
b3nj5m1n/kommentary: README.md

## Configuration

> The config module provides a convenience function called `configure_language` which helps in configuring a language.

Author: ```suggestion
The config module provides a convenience function called `configure_language` which makes it easy to configure a language.
```

### Configure unsupported language

> Most languages have basic support out of the box, thanks to `commentstring`, however for some languages `commentstring` is not set, and `commentstring` supports either single-line or multi-line comments, not both.

Author: ```suggestion
Most languages have basic support out of the box, thanks to `commentstring`.
Unfortunately however, for some languages `commentstring` is not set.
Also, `commentstring` supports either single-line or multi-line comments, not both.
```

For those reasons, you might prefer to properly configure a language. You can do it like this:

```lua
lua << EOF
require('kommentary.config').configure_language("rust", {
    single_line_comment = "//",
    multi_line_comment = {"/*", "*/"},
})
EOF
```

If one of those two is not supported by the language, set the value to false, otherwise the default (`//` for single-line and `{/*,*/}` for multi-line) will be used.

### Always use single/multi-line comments

> Some languages might technically support multi-line comments but have some quirks with them, or maybe you just prefer single-line comments. The easy way to configure this is:

Author: ```suggestion
Some languages might technically support multi-line comments but have some quirks with them, or maybe you just prefer single-line comments. The proper way to configure this is:
```

```lua
lua << EOF
require('kommentary.config').configure_language("rust", {
    prefer_single_line_comments = true,
})
EOF
```

It also works the other way:

```lua
lua << EOF
require('kommentary.config').configure_language("rust", {
    prefer_multi_line_comments = true,
})
EOF
```

> 

Author: ```suggestion

If you set both of the to true, it will use the default.

```
